R/pkg/R/sparkR.R

```diff
@@ -278,3 +278,32 @@ sparkRHive.init <- function(jsc = NULL) {
   assign(".sparkRHivesc", hiveCtx, envir = .sparkREnv)
   hiveCtx
 }
+
+#' Assigns a group ID to all the jobs started by this thread until the group ID is set to a
+#' different value or cleared.
+#'
+#' @param sc existing spark context
+#' @param groupid the ID to be assigned to job groups
+#' @param description description for the the job group ID
+#' @param interruptOnCancel flag to indicate if the job is interrupted on job cancellation
```

Comment:
Yeah @davies point is a good one. We can add an example usage here with something like 
```
 #' @examples
 #'\dontrun{
 #' sc <- sparkR.init()
 # setJobGroup(sc, "group", "some group", TRUE)
 #'}
```